bread/layout/components/form.py

```diff
@@ -389,9 +388,14 @@ def extract_attrs(data):
         or WIDGET_MAPPING[type(field.field.widget)]
     )
     if isinstance(fieldtype, type) and issubclass(fieldtype, hg.BaseElement):
-        ret = fieldtype(
-            fieldname=field.name, widgetattributes=attrs, **elementattributes
-        )
```

Comment:
Adding these if statements all over the place in _mapwidget ist ugly but I think it is exactly the right choice until we have all form-components refactored to use the "new" style. And then we can simplify the logic inside _mapwidget :+1: